app/src/main/java/itaycsguy/rtchordslearningapk/SigninActivity.kt

```diff
@@ -14,140 +14,166 @@ import com.google.android.gms.auth.api.signin.GoogleSignInResult
+//    private lateinit var profile_section: ConstraintLayout
+    private lateinit var signInButton:  SignInButton
+    private lateinit var localSignInButton:  Button
+    private lateinit var localSignUpButton:  Button
+    private lateinit var localForgotButton: Button
+    private lateinit var username: EditText
+    private lateinit var text_username: String
+    private lateinit var password: EditText
+    private lateinit var googleApiClient: GoogleApiClient
+    private var googleName: String? = null
+    private var googleEmail: String? = null
+    private var googleImage: String? = null
+    private var signInOptions: GoogleSignInOptions = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN).requestEmail().build()
+    private val REQ_CODE = 201
+    private val SIGNED_IN: Int = 0
+    private val SIGNED_OUT: Int = 1
+    private val ERROR: Int = 2
+    private val CONNECTED: Int = 3
+
     override fun onConnectionFailed(p0: ConnectionResult) {
         // need still to implement!
     }
 
-    private lateinit var profile_section: ConstraintLayout
-    private lateinit var sign_out_button: Button
-    private lateinit var sign_in_button:  SignInButton
-    private lateinit var email: TextView
-    private lateinit var name: TextView
-    private lateinit var image_profile: ImageView
-    private lateinit var google_api_client: GoogleApiClient
-    private var sign_in_options: GoogleSignInOptions = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN).requestEmail().build()
-    private val REQ_CODE = 200
-
     private fun init(){
-        this.profile_section = findViewById(R.id.main_layout)
-        this.sign_in_button = findViewById(R.id.google_sign_in) as SignInButton
-        this.sign_out_button = findViewById(R.id.google_sign_out) as Button
-        this.email = findViewById(R.id.text_username) as TextView
-        this.sign_in_button.setOnClickListener(this)
-        this.sign_out_button.setOnClickListener(this)
-        this.google_api_client = GoogleApiClient.Builder(this).enableAutoManage(this,this).addApi(Auth.GOOGLE_SIGN_IN_API,this.sign_in_options).build()
-//        this.profile_section.visibility = View.GONE
-    }
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
-
-        this.init()
-        val buttonSignIn = findViewById(R.id.sign_in_button) as Button
-        buttonSignIn.setOnClickListener {
-            val username = (findViewById(R.id.text_username) as EditText).text.toString()
-            val password = (findViewById(R.id.text_password) as EditText).text.toString()
-            this.local_validate(username,password)
+        this.username = findViewById(R.id.text_username) as EditText
+        this.password = findViewById(R.id.text_password) as EditText
+        this.localSignInButton = findViewById(R.id.sign_in_button) as Button
```

Comment:
you shouldn't be needing the "as" Kotlin should be identifying this since the variables has their types set on the late init

Reply:
also pretty sure u dont need the "this."

is this a copied code from somewhere? (and transformed to kotlin) it might the convert since it doesnt always convert perfectly

Reply:
The "this" keyword has written because of the "clean code" format i  am trying to bring